internal/resources/managementcluster/management_cluster_registration_mock_test.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/jarcoal/httpmock"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	clustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	registrationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementcluster"
+	objectmetamodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/objectmeta"
+)
+
+func setupHTTPMocks(t *testing.T) {
+	httpmock.Activate()
+	t.Cleanup(httpmock.Deactivate)
+}
+
+const (
+	https      = "https:/"
+	apiVersion = "v1alpha1"
+	groups     = "managementclusters"
+)
+
+func setUpOrgPolicyEndPointMocks(t *testing.T, endpoint string, resourceName string, kubernetesProviderTypeValue *clustermodel.VmwareTanzuManageV1alpha1CommonClusterKubernetesProviderType) {
+	managementClusterRequest := &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest{}
+
+	managementClusterResponse := &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse{
+		ManagementCluster: &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterManagementCluster{
+			FullName: &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterFullName{
+				Name: resourceName,
+			},
+			Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+				UID: "1234",
+			},
+			Spec: &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterSpec{
+				DefaultClusterGroup:    "default",
+				KubernetesProviderType: kubernetesProviderTypeValue,
+			},
+			Status: &registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterStatus{
+				KubernetesProvider: &clustermodel.VmwareTanzuManageV1alpha1CommonClusterKubernetesProvider{
+					Type: kubernetesProviderTypeValue,
+				},
+				Phase: registrationmodel.NewVmwareTanzuManageV1alpha1ManagementclusterPhase(registrationmodel.VmwareTanzuManageV1alpha1ManagementclusterPhaseDELETING),
+			},
+		}}
+
+	createManagementClusterRegistrationEndpoint := helper.ConstructRequestURL(https, endpoint, apiVersion, groups).String()
+	readManagementClusterRegistrationEndpoint := helper.ConstructRequestURL(https, endpoint, apiVersion, groups, resourceName).String()
+	deleteManagementClusterRegistrationResourceEndpoint := readManagementClusterRegistrationEndpoint
+
+	httpmock.RegisterResponder("POST", createManagementClusterRegistrationEndpoint,
+		bodyInspectingResponder(t, managementClusterRequest, http.StatusOK, managementClusterResponse))
+
+	httpmock.RegisterResponder("GET", readManagementClusterRegistrationEndpoint,
+		bodyInspectingResponder(t, nil, http.StatusOK, managementClusterResponse))
+
+	httpmock.RegisterResponder("DELETE", deleteManagementClusterRegistrationResourceEndpoint, changeStateResponder(
+		func() {
+			httpmock.RegisterResponder("GET", readManagementClusterRegistrationEndpoint,
+				httpmock.NewStringResponder(http.StatusBadRequest, "Not found"))
+		},
+		http.StatusOK,
+		nil))
+}
+
+// nolint: unparam
```

Comment:
nit: any reason we have the lint disabled?

Reply:
Probably bad leftover. Removed this comment.